src/document.jl

```diff
@@ -53,6 +76,17 @@ mutable struct StringDocument{T<:AbstractString} <: AbstractDocument
     metadata::DocumentMetadata
 end
 
+"""
+    StringDocument(str)
+
+A document represented using a UTF8 String stored in RAM
```

Comment:
I think for offline documentation, this should be better as `create a document represented using ...` (Imperative form)